src/frequenz/sdk/timeseries/_formula_engine/_formula_printer.py

```diff
@@ -0,0 +1,220 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+from __future__ import annotations
+import enum
+
+from ._formula_steps import (
+    Adder,
+    Averager,
+    Clipper,
+    ConstantValue,
+    Divider,
+    FormulaStep,
+    MetricFetcher,
+    Multiplier,
+    OpenParen,
+    Subtractor,
+)
+
+class OperatorPrecedence(enum.Enum):
+    """Represents the precedence of an operator."""
+
+    Subtraction = 1
+    Addition = 2
+    Division = 3
+    Multiplication = 4
+    Primary = 9
+
+    def __gt__(self, other: OperatorPrecedence) -> bool:
+        """Return whether the precedence of this operator is greater than the precedence of the other operator.
+
+        Args:
+            other: The other operator (on the right-hand side).
+
+        Returns:
+            bool: Whether the precedence of this operator is greater than the precedence of the other operator.
+        """
+        return self.value > other.value
+
+
+    def __lt__(self, other: OperatorPrecedence) -> bool:
+        return self.value < other.value
+
+    def __le__(self, other: OperatorPrecedence) -> bool:
+        return self.value <= other.value
+
+    @staticmethod
+    def map(operator: str) -> OperatorPrecedence:
+        """Return the precedence of the operator.
+
+        Args:
+            operator: The operator.
+
+        Returns:
+            int: The precedence of the operator.
+        """
+        match operator:
+            case "+":
+                return OperatorPrecedence.Addition
+            case "-":
+                return OperatorPrecedence.Subtraction
+            case "*":
+                return OperatorPrecedence.Multiplication
+            case "/":
+                return OperatorPrecedence.Division
+            case _:
+                return OperatorPrecedence.Primary
```

Comment:
I would name this `from_str()`, as we are using that convention for quantities for example. `map` in Python means mapping using a function.